wgpu-core/src/device/ray_tracing.rs

```diff
@@ -0,0 +1,376 @@
+use std::mem::ManuallyDrop;
+use std::sync::Arc;
+
+#[cfg(feature = "trace")]
+use crate::device::trace;
+use crate::lock::rank;
+use crate::resource::{Fallible, TrackingData};
+use crate::{
+    device::{queue::TempResource, Device, DeviceError},
+    global::Global,
+    id::{self, BlasId, TlasId},
+    lock::RwLock,
+    ray_tracing::{get_raw_tlas_instance_size, CreateBlasError, CreateTlasError},
+    resource, LabelHelpers,
+};
+use hal::AccelerationStructureTriangleIndices;
+use wgt::Features;
+
+impl Device {
+    fn create_blas(
+        self: &Arc<Self>,
+        blas_desc: &resource::BlasDescriptor,
+        sizes: wgt::BlasGeometrySizeDescriptors,
+    ) -> Result<Arc<resource::Blas>, CreateBlasError> {
+        let size_info = match &sizes {
+            wgt::BlasGeometrySizeDescriptors::Triangles { desc } => {
+                let mut entries =
+                    Vec::<hal::AccelerationStructureTriangles<dyn hal::DynBuffer>>::with_capacity(
+                        desc.len(),
+                    );
+                for x in desc {
```

Comment:
Let's give `x` a descriptive name.

```suggestion
                for desc in descriptors {
```

Reply:
I am confident I have found al of these.